### PR TITLE
profile upload

### PR DESCRIPTION
**Pull request by jeremymv2**
### Description

This PR provides action `upload` on resource `compliance_profile`

It requires PR #91 to be merged first.

Only one mode is supported: direct communication with Compliance Server via token||refresh_token.

Usage is based on the audit hash like so:
```
default['audit']['server'] = 'https://compliance-server.test/api'
default['audit']['collector'] = 'chef-compliance'
default['audit']['token'] = nil
default['audit']['refresh_token'] = '20/cgm927J9371DRh8mZ2wp_bXRJxKPW52rtThkwQz5PzhBIcbq6VLB0LsdrfOXv_aBxprU1LTv3aUvHxe2mR-m2A=='
default['audit']['profiles'] = {
  'admin/ssh2' => {
    'source' => '/some/base_ssh.tar.gz'
  },
  'admin/linux2' => {
    'source' => '/some/base_linux.tar.gz'
  }
}
default['audit']['insecure'] = true
```

The goal is to help enable Compliance Profile uploads via automated pipeline workflow. 

### Issues Resolved

N/A

### Check List
- [x] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

**Comment by jeremymv2** [comment]
If PR #91 passes muster, I'll add unit tests and update the README.md with usage info.

**Comment by chris-rock** [comment]
@jeremymv2 super helpful for build cookbooks

File: libraries/profile.rb [hunk 1]
```diff
@@ -141,6 +141,22 @@ class ComplianceProfile < Chef::Resource
...
+        converge_by 'upload compliance profile' do
+          raise 'unable to determine path to compressed profile archive' if path.nil?
+          o, p = normalize_owner_profile
+          node.run_state['compliance'] ||= {}
+
+          if node.run_state['compliance']['access_token']
+            reqpath ="owners/#{o}/compliance/#{p}/tar"
+            url = construct_url(server, reqpath)
+            Chef::Log.info "Upload compliance profile #{o}/#{p} to: #{url}"
+            res = Compliance::HTTP.post_file(url.to_s, node.run_state['compliance']['access_token'], path, insecure)
```
Comment (chris-rock) [inline libraries/profile.rb]: Currently that is not using the upload function for inspec https://github.com/chef/inspec/blob/master/lib/bundles/inspec-compliance/api.rb#L82 Any specific reason?

We need to be aware that the current method is not using the checking mechanisms as available. https://github.com/chef/inspec/blob/master/lib/bundles/inspec-compliance/cli.rb#L94-L167

Do you think we could combine them? I would make to just reuse functions we already have. If that requires refactoring in InSpec, I would go for that.

**Comment by jeremymv2** [comment]
@chris-rock Thanks!  I'll go ahead and add some tests as well as update the docs with the new functionality.  Then this PR will be ready for review.

File: libraries/profile.rb (hunk 1 above)
Comment (jeremymv2) [inline libraries/profile.rb]: @chris-rock  I agree, using `Compliance::API.upload()` is definitely the way to go.  I'll make that change easily.

The error checking would be a nice to have.  I'm thinking that refactoring the `inspec-compliance` bundle to pull the error checking out of  `ComplianceCli` class would be the longer term solution. The tricky part being to make it non `cli` specific and more API friendly.

What if as a first stab, we duplicate some of the rudimentary error checking from above in the `audit` cookbook resource? Like:

1. path exist
2. profile.check
3. profile exists? (overwrite should probably be the default for a build-cookbook scenario?)

Comment (jeremymv2) [inline libraries/profile.rb]: `Compliance::API.upload()` calls `Compliance::HTTP.post_file()` like so:

`res = Compliance::HTTP.post_file(url, config['token'], archive_path, config['insecure'], !config.supported?(:oidc))`

However, when I pass in a config from `Compliance::Configuration.new()`, `!config.supported?(:oidc))` always resolves to `true` then authentication fails, when `false` the call succeeds.

```
pry(#<Audit::Resource::ComplianceProfile action provider>)> !config.supported?(:oidc)
=> true
[10] pry(#<Audit::Resource::ComplianceProfile action provider>)> !config.supported?(:bar)
=> true
[11] pry(#<Audit::Resource::ComplianceProfile action provider>)>
```

**Comment by jeremymv2** [comment]
Closing this PR in favor of #96

**Pull request closed without merging**